apps/communications/dialer/test/marionette/keypad_test.js

```diff
@@ -102,231 +105,270 @@ marionette('Dialer > Keypad', function() {
+        numberElement.selectionStart = numberElement.selectionEnd = 2;
+        numberElement.click();
+      });
 
-    var number = subject.client.findElement(selectors.phoneNumber);
-    number.scriptWith(function (numberElement) {
-      numberElement.selectionStart = numberElement.selectionEnd = 2;
-      numberElement.click();
+      var del = subject.client.findElement(selectors.del);
+      actions.doubleTap(del).perform();
+
+      assert.equal(number.getAttribute('value'), '3');
     });
 
-    var del = subject.client.findElement(selectors.del);
-    actions.doubleTap(del).perform();
+    test('Clearing the number partially by long pressing the delete key',
+    function() {
+      typeNumber();
 
-    assert.equal(number.getAttribute('value'), '3');
-  });
+      var number = subject.client.findElement(selectors.phoneNumber);
+      number.scriptWith(function (numberElement) {
+        numberElement.selectionStart = numberElement.selectionEnd = 2;
+        numberElement.click();
+      });
 
-  test('Clearing the number partially by long pressing the delete key',
-  function() {
-    typeNumber();
+      var del = subject.client.findElement(selectors.del);
+      actions.longPress(del, 1).perform();
 
-    var number = subject.client.findElement(selectors.phoneNumber);
-    number.scriptWith(function (numberElement) {
-      numberElement.selectionStart = numberElement.selectionEnd = 2;
-      numberElement.click();
+      client.waitFor(function() {
+        return (number.getAttribute('value') === '3');
+      });
+      assert.ok(true, 'cleaned the phone number view');
     });
+});
 
-    var del = subject.client.findElement(selectors.del);
-    actions.longPress(del, 1).perform();
+suite('Add dialed number to a contact', function() {
+    function addTestContact(name, telephone) {
+    contacts.addContact({
+      givenName: name,
+      tel: telephone
+    });
+    }
 
-    client.waitFor(function() {
-      return (number.getAttribute('value') === '3');
+    setup(function() {
+      subject = new Dialer(client);
+      
+      var contactName = 'Name';
+      var contactFirstPhone = '12345';
+      
+      contacts = new Contacts(client);
+      contacts.launch();
+      addTestContact(contactName, contactFirstPhone);
+      contacts.close();
     });
-    assert.ok(true, 'cleaned the phone number view');
-  });
+    
+      test('Add dialed number to a contact', function() {
+        var contactSecondPhone = '67890';
+        var contactName = 'Name';
+
+        subject.launch();   
+        subject.keypadView.typePhoneNumber(contactSecondPhone);
 
+        var addToContactsActivities = subject.keypadView.addToContacts();
+        var contact = addToContactsActivities.pickAddToContact();
+        var dialer = 
+          contact.addToContactAndReturnToDialer(contactName);
+        var contactList = dialer.tabs.goToContactList();
+        var contactUpdatedName = contactList.getContactFirstName();
```

Comment:
`contactUpdatedName` sounds like you changed the name of the contact. What do you think of `contactDetailsDisplayedName`?